subprojects/core/src/main/java/org/gradle/initialization/DefaultClassLoaderRegistry.java

```diff
@@ -93,4 +102,23 @@ public ClassLoader getGradleCoreApiClassLoader() {
     public MixInLegacyTypesClassLoader.Spec getGradleWorkerExtensionSpec() {
         return workerExtensionSpec;
     }
+
+    @Override
+    public VisitableURLClassLoader.Spec getUserSpec(String name, Iterable<File> additionalClasspath, Class<?>... classes) {
```

Comment:
Feels like we should pull this stuff out into its own service (at least user spec and worker extension spec).  ApiFilterSpec was added here because it was hard to disentangle it from the plugin classloaders without doing more awkward things, and then worker extension spec was added later, but I wonder if we should just have a separate spec registry or something...

Reply:
I can just fold this back into ClassLoaderStructureProvider since it's specific to the workers for now.